File: 75. Sort Colors/75.py
class Solution:
    def sortColors(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        low = 0
        mid = 0
        hight = len(nums) -1

        while mid<=hight:

            if nums[mid] == 0:

                aux = nums[low]
                nums[low] = nums[mid]
                nums[mid] = aux

                low+=1
                mid+=1
            
            elif nums[mid] == 1:
                mid+=1

            else:
                aux = nums[hight]
                nums[hight] = nums[mid]
                nums[mid] = aux

                hight -=1

            print(nums)

S = Solution()
print(S.sortColors([2,0,1]))

File: 75. Sort Colors/75.cs
public class Solution {
    public void SortColors(int[] nums) {

        int low = 0;
        int mid = 0;
        int hig = nums.Length - 1;

        while(mid <= hig)
        {
            if (nums[mid] == 0)
            {
                int temp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = temp;

                low++;
                mid++;

            }
            else if (nums[mid] == 1)
            {
                mid++;
            }
            else
            {
                int temp = nums[hig];
                nums[hig] = nums[mid];
                nums[mid] = temp;

                hig--;
            }

        }
        
    }
}